fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java

```diff
@@ -1739,6 +1745,13 @@ public PlanFragment visitPhysicalSetOperation(
             setPlanRoot(setOperationFragment, setOperationNode, setOperation);
         }
 
+        // in pipeline engine, we use parallel scan by default, but it broke the rule of data distribution
+        // we need turn of parallel scan to ensure to get correct result.
+        // TODO: nereids forbid all parallel scan under PhysicalSetOperation temporary
+        if (findOlapScanNodesByPassExchangeAndJoinNode(setOperationFragment.getPlanRoot())) {
+            setOperationFragment.setHasColocatePlanNode(true);
+        }
```

Comment:
check set operation output physical properties, if it is any, do not set this flag